src/semaphore_internal.h

```diff
@@ -29,6 +29,13 @@
 
 struct dispatch_queue_s;
 
+#if USE_FUTEX_SEM
+typedef struct {
+	/* Low 32 bits are the count; high bits are the waiter count. */
+	volatile uint64_t dfx_data;
+} dispatch_futex_t;
+#endif  /* USE_FUTEX_SEM */
```

Comment:
The problem is that I have to reload it .. Let me massage a bit again, will see how it looks like.
Wanna move on to other problems ..